### ISSUE TITLE
Contracts.Precondition.Not.Satisfied error

### ISSUE THREAD

**Issue by aagarwal1012** [typetools/checker-framework#2408]
I got the `contracts.precondition.not.satisfied` error while running the following code:
**[Link](http://eisop.uwaterloo.ca/live/#mode=display&code=import+org.checkerframework.checker.nullness.qual.Nullable%3B%0Aimport+org.checkerframework.checker.nullness.qual.NonNull%3B%0Aimport+org.checkerframework.checker.nullness.qual.RequiresNonNull%3B%0Aclass+A+%7B%0A++%0A++%40NonNull+String+message+%3D+%22default+string%22%3B%0A++%0A++%40RequiresNonNull(%22%231%22)%0A++void+setMessage(String+message)%7B%0A++++this.message+%3D+message%3B%0A++%7D%0A++%0A++String+getMessage()%7B%0A++++return+message%3B%0A++%7D%0A++%0A++void+main()%7B%0A++++A+object+%3D+new+A()%3B%0A++++String+mess+%3D+%22random+message%22%3B%0A++++object.setMessage(mess.substring(0,+mess.length()+-+3))%3B%0A++%7D+%0A%7D&typeSystem=nullness)** to the code.
```java
import org.checkerframework.checker.nullness.qual.Nullable;
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.RequiresNonNull;
class A {
  
  @NonNull String message = "default string";
  
  @RequiresNonNull("#1")
  void setMessage(String message){
    this.message = message;
  }
  
  String getMessage(){
    return message;
  }
  
  void main(){
    A object = new A();
    String mess = "random message";
    object.setMessage(mess.substring(0, mess.length() - 3));
  } 
}
```
The exact error message follows :
I had used the Checker Framework Live Demo webpage to produce the error.
```bash
Error: [contracts.precondition.not.satisfied] the called method 'object.setMessage(mess.substring(0, mess.length() - 3))' has a precondition 'mess.substring(0, ?)' that is not satisfied
```

**Comment by mernst** [comment]
You are right that this is an inconsistency between treatment of the two types of annotations.  Thanks for noticing this.  This is due to details of the implementation and is unlikely to be fixed in the short term.

In the short term, we can make the Checker Framework issue a warning (not an error) if the user writes `@Requires*` whose argument is exactly a formal parameter name such as "this" or "#1" or "#3".  The Checker Framework gives better diagnostics for those.  When we have done so, we will close this issue.
